app/src/main/java/org/wikipedia/dataclient/page/PageSummary.kt

```diff
@@ -0,0 +1,106 @@
+
+    @SerializedName("originalimage")
+    private val originalImage: Thumbnail? = null
+
+    @SerializedName("wikibase_item")
+    val wikiBaseItem: String? = null
+
+    @SerializedName("extract_html")
+    val extractHtml: String? = null
+
+    @SerializedName("description_source")
+    val descriptionSource = ""
+
+    private var thumbnail: Thumbnail? = null
+    private val namespace: NamespaceContainer? = null
+    private var titles: Titles? = null
+    var lang = ""
+    var extract: String? = null
+    var description: String? = null
+
+    @JsonAdapter(GeoTypeAdapter::class)
+    val geo: Location? = null
+    val type = TYPE_STANDARD
+    val pageId = 0
+    val revision = 0L
+    val timestamp = ""
+    val thumbnailUrl: String?
+        get() = if (thumbnail == null) null else thumbnail!!.url
+    val thumbnailWidth: Int
+        get() = if (thumbnail == null) 0 else thumbnail!!.width
+    val thumbnailHeight: Int
+        get() = if (thumbnail == null) 0 else thumbnail!!.height
+    val originalImageUrl: String?
+        get() = originalImage?.url
+    val apiTitle: String
+        get() = StringUtils.defaultString(if (titles != null) titles!!.canonical else null)
+
+    // TODO: Make this return CharSequence, and automatically convert from HTML.
+    val displayTitle: String
+        get() = StringUtils.defaultString(if (titles != null) titles!!.display else null)
+
+    constructor()
+    constructor(displayTitle: String, prefixTitle: String, description: String?,
+                extract: String?, thumbnail: String?, lang: String) {
+        titles = Titles(prefixTitle, displayTitle)
+        this.description = description
+        this.extract = extract
+        this.thumbnail = Thumbnail(thumbnail, 0, 0)
+        this.lang = lang
+    }
+
+    fun toPage(title: PageTitle): Page {
+        return Page(adjustPageTitle(title), PageProperties(this))
+    }
+
+    private fun adjustPageTitle(title: PageTitle): PageTitle {
+        var newTitle = title
+        if (titles != null && titles!!.canonical != null) {
+            newTitle = PageTitle(titles!!.canonical, title.wikiSite, title.thumbUrl)
+            newTitle.fragment = title.fragment
+        }
+        newTitle.description = description
+        return newTitle
+    }
+
+    fun getNamespace(): Namespace {
+        return if (namespace == null) Namespace.MAIN else Namespace.of(namespace.id)
+    }
+
+    fun getPageTitle(wiki: WikiSite): PageTitle {
+        return PageTitle(apiTitle, wiki, thumbnailUrl, description, displayTitle, extract)
+    }
+
+    private class Thumbnail(val url: String?, val width: Int, val height: Int)
+
+    private class NamespaceContainer {
+        val id = 0
+    }
+
+    private class Titles(val canonical: String?, val display: String?)
+
+    override fun toString(): String {
+        return displayTitle
+    }
+
+    val leadImageName: String?
+        get() = if (thumbnailUrl == null) null else getFilenameFromUploadUrl(thumbnailUrl!!)
```

Comment:
Can be simplified to `val leadImageName get() = thumbnailUrl?.let { getFilenameFromUploadUrl(it) }`